src/reanimated2/animation/timing.ts

```diff
@@ -65,6 +65,29 @@ export const withTiming = function (
 ): Animation<TimingAnimation> {
   'worklet';
 
+  if (__DEV__) {
+    // validate config
+    if (userConfig?.easing) {
+      const { easing } = userConfig;
+      const isFunction: boolean = typeof easing === 'function';
+      const isWorklet: boolean = !!(
+        easing as unknown as { __workletHash: number }
+      )?.__workletHash;
+      const functionName = typeof easing === 'function' ? easing?.name : '';
+      /** When calling `withTiming` on UI we don't pass worklets directly, but we
+      bound them first. This happens, for example, when animating with keyframes */
+      const isBound = functionName.startsWith('bound');
```

Comment:
Why exactly do we need to check that function is not bound?

Reply:
@tomekzaw I described it in my comment.

Reply:
can't find the comment :(

Reply:
Here you go! 
![Screenshot 2024-02-12 at 17 21 00](https://github.com/software-mansion/react-native-reanimated/assets/40713406/dcab8f43-0864-4a80-8c44-654ff62e0914)

Reply:
Aaaa, this comment, thanks. What about bound functions in the RN runtime?

Reply:
Now when I think about it, I think it's enough to just check for `isWorklet` here, isn't it?

Reply:
Probably yes

Reply:
Actually it's not sufficient in case of Keyframe animations - however I think understanding why is that should be a separate task (if we actually want to investigate it).